### PR TITLE
Stocks Fix

### LINKED ISSUE #473: Stocks 3.0 Filter - Show only owned stock blocks

### PR DESCRIPTION
**Pull request by Sashank999**
Closes #473 .

### CODE REVIEW

File: torntools/scripts/content/stockexchange/ttStockexchange.js [hunk 1]
```diff
@@ -232,191 +48,121 @@ function addFilter(filters) {
...
-				? stock
-						.find(".logo > a")
-						.getAttribute("href")
-						.match(/&ID=([0-9]*)/i)[1]
-				: stock.firstElementChild.getAttribute("action").split("ID=")[1];
-
-			const data = torndata.stocks[stockId];
+		for (let stock of doc.findAll("#stockmarketroot [class*='stockMarket__'] > *:not(#panel-InfoTab, #panel-ManagerTab)")) {
+			const stockAcronym = stock.find("[class*='logoContainer__'] img").src.split("/").pop().replace(".svg", "");
+			const stockID = torndata.stocks.findIndex((stockEntry) => {
```
Comment (DeKleineKobini) [inline torntools/scripts/content/stockexchange/ttStockexchange.js:150]: They changed the stocks api, so this is no longer correct.

File: torntools/scripts/content/stockexchange/ttStockexchange.js [hunk 2]
```diff
@@ -232,191 +48,121 @@ function addFilter(filters) {
...
+				return stockEntry.acronym === stockAcronym;
+			});
+			const data = torndata.stocks[stockID];
+			const userStockData = userdata.stocks.filter((x) => {
```
Comment (DeKleineKobini) [inline torntools/scripts/content/stockexchange/ttStockexchange.js:154]: They changed the stocks api, so this is no longer correct.

**Review by DeKleineKobini — CHANGES_REQUESTED** [review]

File: torntools/scripts/content/stockexchange/ttStockexchange.js (hunk 1 above)
Comment (Sashank999) [reply torntools/scripts/content/stockexchange/ttStockexchange.js]: Done.

File: torntools/scripts/content/stockexchange/ttStockexchange.js (hunk 2 above)
Comment (Sashank999) [reply torntools/scripts/content/stockexchange/ttStockexchange.js]: Done.

**Review by DeKleineKobini — APPROVED** [review]

**Pull request merged**